session/session_test.go

```diff
@@ -2162,8 +2162,24 @@ func (s *testSchemaSerialSuite) TestSchemaCheckerTempTable(c *C) {
 
 	// It's fine to change the schema of temporary tables.
 	tk.MustExec(`begin;`)
-	tk1.MustExec(`alter table temp_table modify column c bigint;`)
+	tk1.MustExec(`alter table temp_table modify column c tinyint;`)
+	tk.MustExec(`insert into temp_table values(3, 3);`)
+	tk.MustExec(`commit;`)
+
+	tk.MustExec("begin pessimistic")
+	tk1.MustExec(`alter table temp_table modify column c smallint;`)
 	tk.MustExec(`insert into temp_table values(3, 3);`)
+	tk.MustExec(`select * from temp_table for update;`)
```

Comment:
Try to select both temp table and normal table in one statement.

Reply:
Done